testing/integration/upgrade_test.go

```diff
@@ -162,50 +176,72 @@ func TestStandaloneUpgrade(t *testing.T) {
 		Sudo:    true, // requires Agent installation
 	})
 
-	agentFixture, err := define.NewFixture(t, define.Version())
-
-	require.NoError(t, err)
-
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	err = agentFixture.Prepare(ctx)
-	require.NoError(t, err, "error preparing agent fixture")
 
-	err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
-	require.NoError(t, err, "error configuring agent fixture")
+	versionList := getUpgradableVersions(ctx, t, define.Version())
+
+	for _, v := range versionList {
+		parsedVersion, err := version.ParseVersion(v)
+		if err != nil {
+			t.Logf("version %q returned by artifact api is not parseable: %v", v, err)
```

Comment:
Should this be a `t.Errorf`? I realize that will fail the test but maybe that's what we want? Is there any legitimate reason to receive non-parseable versions from the artifact API?

My concern is that we might hit this error, it will get logged, but the test won't fail so we'll never bother to go and check what's happening. If we make the test fail here, maybe we see some false positives but then we can try to account for those specific cases and continue to fail otherwise.

Reply:
This would fail the whole Standalone upgrade test because of a single version when all the other subtests may be green :thinking: 
Also it cannot really happen as the `getUpgradableVersions()` already parsed the returned versions... At this point I would prefer to switch from a `[]string` to a `[]version.ParsedSemVer`

Reply:
> At this point I would prefer to switch from a `[]string` to a `[]version.ParsedSemVer`

This is a good point.  I looked at the two places `getUpgradableVersions()` is being called from and in both cases, the returned versions (`string`s) eventually get parsed into `version.ParsedSemVer`s.  And it appears that within `getUpgradableVersions()` we are going from `version.ParsedSemVer` to `string` to build the `[]string` that's returned.  So maybe `getUpgradableVersions()` should just return `[]version.ParsedSemVer` instead?